tests/test_ctl.py

```diff
@@ -40,107 +45,196 @@ def test__wait_args_to_opts():
+
+
+@pytest.mark.parametrize(
+    "rc, status",
+    [
+        [0, ctl.Status.running],
+        [3, ctl.Status.not_running],
+        [4, ctl.Status.unspecified_datadir],
+    ],
+)
+def test_status(pgctl: ctl.PGCtl, rc: int, status: ctl.Status) -> None:
+    with patch.object(
+        pgctl, "run_command", return_value=subprocess.CompletedProcess([], rc)
+    ) as run_command:
+        actual = pgctl.status("data")
+    run_command.assert_called_once_with(["pg_ctl", "status", "-D", "data"])
+    assert actual == status
+
 
+def test_status_rc1(pgctl: ctl.PGCtl) -> None:
+    with patch.object(
+        pgctl, "run_command", return_value=subprocess.CompletedProcess([], 1)
+    ) as run_command, pytest.raises(subprocess.CalledProcessError):
+        pgctl.status("data")
+    run_command.assert_called_once_with(["pg_ctl", "status", "-D", "data"])
 
-def test_status(fake_pgctl):
-    assert fake_pgctl.status("data") == ctl.Status.running
-    fake_pgctl.code = 3
-    assert fake_pgctl.status("data") == ctl.Status.not_running
-    fake_pgctl.code = 4
-    assert fake_pgctl.status("data") == ctl.Status.unspecified_datadir
-    fake_pgctl.code = 1
-    with pytest.raises(subprocess.CalledProcessError):
-        fake_pgctl.status("data")
+
+@pytest_asyncio.fixture
+async def apgctl(bindir: Path) -> ctl.AsyncPGCtl:
+    async def run_command(args: Sequence[str], **kwargs: Any) -> ctl.CompletedProcess:
+        return run_command_version_only(args, **kwargs)
+
+    c = await ctl.AsyncPGCtl.get(bindir, run_command=run_command)
+    c.pg_ctl = Path("pg_ctl")
+    return c
+
+
+@pytest.mark.asyncio
+async def test_version_async(apgctl: ctl.AsyncPGCtl) -> None:
+    assert apgctl.version == 110010
+
+
+@pytest.mark.parametrize(
+    "rc, status",
+    [
+        [0, ctl.Status.running],
+        [3, ctl.Status.not_running],
+        [4, ctl.Status.unspecified_datadir],
+    ],
+)
+@pytest.mark.asyncio
+async def test_status_async(
+    apgctl: ctl.AsyncPGCtl, rc: int, status: ctl.Status
+) -> None:
+    with patch.object(
+        apgctl, "run_command", return_value=subprocess.CompletedProcess([], rc)
+    ) as run_command:
+        actual = await apgctl.status("data")
+    run_command.assert_called_once_with(["pg_ctl", "status", "-D", "data"])
+    assert actual == status
+
+
+@pytest.mark.asyncio
+async def test_status_rc1_async(apgctl: ctl.AsyncPGCtl) -> None:
+    with patch.object(
+        apgctl, "run_command", return_value=subprocess.CompletedProcess([], 1)
+    ) as run_command, pytest.raises(subprocess.CalledProcessError):
+        await apgctl.status("data")
+    run_command.assert_called_once_with(["pg_ctl", "status", "-D", "data"])
 
 
 @pytest.fixture(scope="module")
-def pg_ctl():
+def pg_ctl() -> ctl.PGCtl:
     try:
         return ctl.PGCtl()
     except OSError as e:
         pytest.skip(str(e))
 
 
+@pytest_asyncio.fixture(scope="module")
+async def apg_ctl() -> ctl.AsyncPGCtl:
```

Comment:
So we now have 2 fixtures: `apgctl` and `apg_ctl`. It's not very easy to understand when to use which.

Reply:
Indeed, I'm splitting the test file to isolate functional tests.